alf/algorithms/vae.py

```diff
@@ -0,0 +1,81 @@
+# Copyright (c) 2019 Horizon Robotics. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import tensorflow as tf
+
+
+class VariationalAutoEncoder:
```

Comment:
need to be subclass of tf_agents.networks.Network

Reply:
not sure about the benefits of deriving from that class, the current way offer more flexibilities of combining with other modules

Reply:
In many algorithms, we need the ability to clone a model (e.g. target network for dqn). That class make this possible.